interface.go

```diff
@@ -16,6 +16,16 @@ type StreamID = protocol.StreamID
 // A VersionNumber is a QUIC version number.
 type VersionNumber = protocol.VersionNumber
 
+// Some known QUIC versions.
+// Warning: This API should not be considered stable and might change soon.
+const (
+	Version39  VersionNumber = protocol.Version39
+	VersionTLS VersionNumber = protocol.VersionTLS
```

Comment:
I'm not a fan of exposing the VersionTLS here as it's super experimental.

Reply:
Should I remove the `VersionNumber` constants and the `SupportedVersions` altogether?

Reply:
It seems like the only way to completely disentangle the examples from internal packages would be to declare `quic.VersionTLS` (in order to support the tls command line flag). I definitely don't want to do that, `VersionTLS` permanently changes and should under no circumstances be used.

While I generally agree that the example should be copy-pastable, I'm using -tls (as well as -v) **very** heavily in my development workflow, so I'm not really happy to merge anything that makes my life harder.

Reply:
If we disentangle it, you could easily copy the example and adjust it to your needs 😃
How about making the TLS Version (is this really just _TLS_ or should it be the _IETF_ Version?) also available via an environment var?

Reply:
This will at some point be the IETF version, at the moment it's a constantly changing mix between gQUIC and IETF QUIC.
There's no need for this change, if we don't make using the example server and client more complicated than it is today.

Reply:
I'm not sure I understand the issue here. You can always use an int in the versions field in the config, and still have the tls flag, even without the const being exported?

Reply:
I thought the whole point was to make the code better. Using a plain int achieves quite clearly the opposite, especially since the number I often change TLS version number during interop testing.